src/hooks/sync-sites/use-site-sync-management.ts

```diff
@@ -77,8 +78,13 @@ export const useSiteSyncManagement = ( {
 				);
 				const sitesToConnect = [ site, ...stagingSites ];
 
-				const newConnectedSites = await getIpcApi().connectWpcomSite( sitesToConnect, localSiteId );
-				setConnectedSites( newConnectedSites );
+				const newConnectedSites = await getIpcApi().connectWpcomSite(
+					sitesToConnect,
+					localSiteIdToConnect
+				);
+				if ( localSiteIdToConnect === localSiteId ) {
+					setConnectedSites( newConnectedSites );
+				}
```

Comment:
We only update the state if the "selectedSite.id" is the original site that triggered the flow.

In the **else** clause we could focus to the local site, as a possible follow-up UX improvement, we could switch to the local site to the selected site and the Sync tab. Or maybe we could display a notification informing the user.

Reply:
A possible improvement could be to refactor this hook state and pass the localSiteId to `setConnectedSites` and make sure it always updates the correct site.